packages/neon-domain/src/provider/common.ts

```diff
@@ -0,0 +1,27 @@
+import { logging, rpc, sc, u } from "@cityofzion/neon-core";
+
+export interface DomainProvider {
+  name: string;
+  resolveDomain(url: string, domain: string, tld: string): Promise<string>;
+}
+const log = logging.default("neon-domain");
+
+export async function resolve(

```

Comment:
I dont think we can define this as common functionality. Let's move it back under the nns domain.